docs/source/advanced_usage.rst

```diff
@@ -32,6 +32,84 @@ locally):
 - QVM URL: ``http://127.0.0.1:5000``
 - quilc URL: ``tcp://127.0.0.1:5555``
 
+Concurrency
+~~~~~~~~~~~
+
+Using pyQuil for concurrent programming is as simple as calling :py:func:`~pyquil.get_qc()` from within a given context
+(be it a thread or process), then using the returned :py:class:`~pyquil.api.QuantumComputer` as usual within that
+context. While :py:class:`~pyquil.api.QuantumComputer` objects as a whole are not safe to share between contexts (due to
+state related to currently-running compilation or execution requests), some information they use is. Information related
+to client configuration (:py:class:`~pyquil.api.QCSClientConfiguration`) and QPU auth
+(:py:class:`~pyquil.api.EngagementManager`) can be safely extracted and shared among
+:py:class:`~pyquil.api.QuantumComputer` instances, as shown below, to save your code from redundant disk reads and
+auth-related HTTP requests.
+
+Using Multithreading
+--------------------
+
+.. code:: python
+
+    from multiprocessing.pool import ThreadPool
+
+    from pyquil import get_qc
+    from pyquil.api import EngagementManager, QCSClientConfiguration
+    from pyquil.quil import Program
+
+
+    configuration = QCSClientConfiguration.load()
+    engagement_manager = EngagementManager(client_configuration=configuration)
+
+
+    def run(program: Program):
+        qc = get_qc("Aspen-8", client_configuration=configuration, engagement_manager=engagement_manager)
+        return qc.run(qc.compile(program))
+
+
+    programs = [Program("DECLARE ro BIT", "RX(pi) 0", "MEASURE 0 ro").wrap_in_numshots_loop(10)] * 20
+    with ThreadPool(5) as pool:
+        results = pool.map(run, programs)
+
+    for i, result in enumerate(results):
+        print(f"Results for program {i}:\n{result}\n")
+
+
+Using Multiprocessing
+---------------------
+
+.. code:: python
+
+    from multiprocessing.pool import Pool
+
+    from pyquil import get_qc
+    from pyquil.api import EngagementManager, QCSClientConfiguration
+    from pyquil.quil import Program
+
+
+    configuration = QCSClientConfiguration.load()
+    engagement_manager = EngagementManager(client_configuration=configuration)
+
+
+    def run(program: Program):
+        qc = get_qc("Aspen-8", client_configuration=configuration, engagement_manager=engagement_manager)
+        return qc.run(qc.compile(program))
+
+
+    programs = [Program("DECLARE ro BIT", "RX(pi) 0", "MEASURE 0 ro").wrap_in_numshots_loop(10)] * 20
+    with Pool(5) as pool:
+        results = pool.map(run, programs)
+
+    for i, result in enumerate(results):
+        print(f"Results for program {i}:\n{result}\n")
+
+.. note::
+    If you encounter error messages on macOS similar to the following:
+
+    .. parsed-literal::
+        +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
+
+    try setting the environment variable ``OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES``.
+
```

Comment:
maybe a note here saying that if you're targetting a QPU then those requests will be executed serially (and without deterministic order!). if you're targeting a QVM, then they will be executed in parallel

Reply:
Done.